### ISSUE TITLE
[Bug Report][2.5.10] v-data-table fixed-header prop results in gap in top border of built in v-data-footer

### ISSUE THREAD

**Issue by eabidorcls** [vuetifyjs/vuetify#14370]
### Environment
**Vuetify Version:** 2.5.10
**Vue Version:** 2.6.11
**Browsers:** Chrome 95.0.4638.69
**OS:** Windows 10

### Steps to reproduce
Create a v-data-table, populate with data and rows, make sure built-in v-data-footer is visible. If adding fixed-header boolean prop to v-data-table, a small gap appears on the right side of the light grey border on top of the footer element. It appears to be caused by a newly added 17px right margin on the v-data-footer element, which appears to be for accommodating a scroll bar which does not exist (except if height prop is defined and table row height exceeds this height)

### Expected Behavior
There should be a light grey 1px border across the entire v-data-table element above the built in v-data-footer element

### Actual Behavior
A small visual gap, 17px in my env, appears on the right hand side of the border above the built in v-data-footer element. 

### Reproduction Link
<a href="https://codepen.io/eabidor/pen/pordXJG?editors=101" target="_blank">https://codepen.io/eabidor/pen/pordXJG?editors=101</a>

**Comment by KareemDa** [comment]
It doesn't seem that sticky header is working at all!
I've tried to add more rows to make the table scroll. but the header is not fixed

**Comment by MajesticPotatoe** [comment]
> It doesn't seem that sticky header is working at all! I've tried to add more rows to make the table scroll. but the header is not fixed

**fixed-header** requires an explicit height to be operational

The gap is caused by the margin placed on the footer in anticipation of the table body scrollbar (though in your example, because there is no overflow, the scollbar is not actively visible, so you end up with the margin just pushing the border on the footer over.

easy solution likely is to convert `margin-right: 17px` to `padding-right: 25px`

**Comment by KareemDa** [comment]
okay, I'll fix it and make a PR.

**Comment by eabidorcls** [comment]
Yes, @MajesticPotatoe described the issue better than me. I should have mentioned that when I found the issue, the height prop was defined, and tied to a dynamic variable that increased proportionally with the number of rows in the table increased, until it reached a predefined max pixel height. So the fixed header would work properly only after this max-height was reached, and the overflow kicked in and added a scroll bar. Before the max height was reached though, the margin-right was added but the scroll bar was not, causing the visual gap in the border. Should have probably added that aspect to the demo code.
